### PR TITLE
provider/aws: Convert AWS ELB to aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
This converts AWS ELB resource to use aws-sdk-go, removing `goamz/elb`.

### CODE REVIEW

File: builtin/providers/aws/structure_test.go [hunk 1]
```diff
@@ -312,3 +312,22 @@ func Test_flattenParameters(t *testing.T) {
 		}
 	}
 }
+
+func Test_expandInstanceString(t *testing.T) {
```
Comment (mitchellh) [inline builtin/providers/aws/structure_test.go]: Should be named `TestExpandInstanceString` for idiomatic reasons.

**Comment by mitchellh** [comment]
LGTM, another simpler one? :)

**Comment by catsby** [comment]
> LGTM, another simpler one? :)

Mostly; took some time to track down a nil pointer dereference that got introduced into `flattenListeners`:

- https://github.com/hashicorp/terraform/pull/1090/files#diff-0346d6dbe5a0ec0a5a79addaab270bcdR193

Hooray for `aws.String()`s

File: builtin/providers/aws/structure_test.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/structure_test.go]: Are the other tests in that file not idiomatic then? I can covert them:

```console
$ more builtin/providers/aws/structure_test.go | grep 'func Test'
func Test_expandIPPerms(t *testing.T) {
func Test_flattenIPPerms(t *testing.T) {
func Test_expandListeners(t *testing.T) {
func Test_flattenHealthCheck(t *testing.T) {
func Test_expandStringList(t *testing.T) {
func Test_expandParameters(t *testing.T) {
func Test_flattenParameters(t *testing.T) {
```

Comment (phinze) [inline builtin/providers/aws/structure_test.go]: Yeah - as I've learned it, the pattern is: `TestMethodName_specialCase`

**Pull request merged**